app/V1Module/presenters/ExercisesConfigPresenter.php

```diff
@@ -684,28 +685,36 @@ public function checkSetScoreConfig(string $id)
     /**
      * Set score configuration for exercise.
      * @POST
+     * @Param(type="post", name="scoreCalculator", validation="string", description="ID of the score calculator")
      * @Param(type="post", name="scoreConfig", validation="string", description="A configuration of the score calculator (the exact format depends on the calculator assigned to the exercise)")
      * @param string $id Identifier of the exercise
      * @throws ExerciseConfigException
      */
     public function actionSetScoreConfig(string $id)
     {
         $exercise = $this->exercises->findOrThrow($id);
+        $oldConfig = $exercise->getScoreConfig();
 
         $req = $this->getRequest();
+        $calculatorName = $req->getPost("scoreCalculator");
         $config = $req->getPost("scoreConfig");
 
         // validate score configuration
-        $calculator = $this->calculators->getCalculator($exercise->getScoreCalculator());
-        $normalizedConfig = $calculator->validateAndNormalizeScore($config);  // throws if validation fails
+        $calculator = $this->calculators->getCalculator($calculatorName);
+        $normalizedConfig = $config !== null
```

Comment:
I think, that in this case, better approach would be to `validateAndNormalizeScore` even for nulls. In current implementation you can have `weighted` calculator config with `null` value, which should be ideally avoided.

In uniform `validateAndNormalizeScore` you can easily check that the config is null and in weighted that the value is not null. 

Additionally in `WeightedScoreCalculator.getTestWeights(string)` method is non-nullable `string` parameter. But this method is called from `WeightedScoreCalculator.isScoreConfigValid(?string)` which has nullable string parameter...